tests/core/asyncio/test_broadcast.py

```diff
@@ -0,0 +1,77 @@
+import asyncio
+from pathlib import (
+    Path,
+)
+import tempfile
+import uuid
+
+import pytest
+
+from lahja import (
+    BaseEvent,
+    ConnectionConfig,
+    Endpoint,
+)
+
+
+@pytest.fixture
+def config():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        name = str(uuid.uuid4())
+        ipc_path = Path(temp_dir) / name
+        yield ConnectionConfig(name, ipc_path)
+
+
+@pytest.fixture(scope='function')
+async def server(config):
+    # TODO< verify with test client.broadcast() -> server and server.broadcast() -> client works
+    endpoint = Endpoint()
+    await endpoint.start_serving(config)
+    # We need to connect to our own Endpoint if we care about receiving
+    # the events we broadcast. Many tests use the same Endpoint for
+    # broadcasting and receiving which is a valid use case so we hook it up
+    with endpoint:
+        yield endpoint
+
+
+@pytest.fixture(scope='function')
+async def client(server, config):
```

Comment:
TODO, drop these in favor of the `endpoint_pair` fixture that automatically tests both directions.